src/KMonad/Args/Parser.hs

```diff
@@ -367,11 +367,25 @@ defaliasP = many $ (,) <$> lexeme word <*> buttonP
 --------------------------------------------------------------------------------
 -- $defsrc
 
-defsrcP :: Parser DefSrc
-defsrcP = many $ lexeme keycodeP
+newSettings :: Monoid s => ASetter' s [a] -> a -> s
+newSettings field value = set field [value] mempty
 
+defsrcP :: Parser DefSrc
+defsrcP = let f s v p = keywordP s $ newSettings v <$> p in
+  fmap fold . many . lexeme . choice . map try $
+    [ f "name" name word
+    , newSettings keycodes <$> keycodeP
+    ]
```

Comment:
This seems a bit "unnecessarily general", I think. To me it seems more sensible to restrict `defsrc` blocks to only have one name, specified right after the `defsrc` keyword (this would incidentally also make the implementation more straightforward)

Reply:
I used it since when I implemented it I also added `fallthrough` as an option. Removing it would simplify the code though it would be readded when imlpmenting other options. Furthermore it adds the ability to pick where you want to add it (start, end, or even in the middle).
Could still remove it if code simplicity is the goal.

Reply:
> Removing it would simplify the code though it would be readded when imlpmenting other options.

We can certainly re-add it when other options come into play, but this does not seem especially relevant for now.

> Furthermore it adds the ability to pick where you want to add it (start, end, or even in the middle). This is nice since in ather cases it's optionally at the end.

I think that this is actually a disadvantage—don't want to give people too much room to mess things up :) Having it always be at the start of the expressions seems fine to me.